internal/app.go

```diff
@@ -209,29 +211,49 @@ func (a *App) extract(crypto CryptoHandler) error {
 	if _, err := os.Stat(a.SecretsDir); err != nil {
 		return err
 	}
-	config, err := Unmarshall(crypto, a.EncryptedConfig)
+	config, err := Unmarshall(crypto, a.EncryptedConfig, true)
 	if err != nil {
 		return err
 	}
 
+	all_fname := make(map[string]bool)
 	for fname, cfgFile := range config {
 		log.Printf("Extracting %s", fname)
+		all_fname[fname] = true
 		fullpath := filepath.Join(a.SecretsDir, fname)
 		changed, err := updateSecret(fullpath, []byte(cfgFile.Value))
 		if err != nil {
 			return err
 		}
-		if changed && len(cfgFile.OnChanged) > 0 {
-			log.Printf("Running on-change command for %s: %v", fname, cfgFile.OnChanged)
-			cmd := exec.Command(cfgFile.OnChanged[0], cfgFile.OnChanged[1:]...)
-			cmd.Env = append(os.Environ(), "CONFIG_FILE="+fullpath)
-			cmd.Stdout = os.Stdout
-			cmd.Stderr = os.Stderr
-			if err := cmd.Run(); err != nil {
-				log.Printf("Unable to run command: %v", err)
-			}
+		if changed {
+			runOnChanged(fname, fullpath, cfgFile.OnChanged)
+		}
+	}
+
+	// Now, watch for file removals (compare with previous version)
+	if _, err := os.Stat(a.EncryptedBackup); err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		} else {
+			return err
+		}
+	}
+	prev, err := Unmarshall(nil, a.EncryptedBackup, false)
+	if err != nil {
+		return err
+	}
+	for fname, cfgFile := range prev {
+		if _, ok := all_fname[fname]; ok {
+			continue
 		}
+		log.Printf("Removing %s", fname)
+		fullpath := filepath.Join(a.SecretsDir, fname)
+		if err := os.Remove(fullpath); err != nil {
+			return err
```

Comment:
Based on @doanac 's comment, I deduct that a file might already be not present on a system (e.g. after a reboot).
For a proper behavior we should not fail for a missing file, but still run onchanged handlers.